src/components/game/index.jsx

```diff
@@ -1,9 +1,77 @@
-import React from "react";
+import React, { useState } from "react";
 import Board from "components/game/Board";
 
 // Renders the Board component
 function Game() {
-  return <Board />;
+  const [valuesArr, setValuesArr] = useState(Array(9).fill(null)); // Values of the 1st-9th square
+  const [nextPlayer, setNextPlayer] = useState("O"); // Initial player aka first player to go is O
+  const [winner, setWinner] = useState(""); // Winner is initially blank
+
+  // i parameter - index of the valuesArr we want to change aka the Nth square
+  function updateValuesArr(i) {
+    // If the Square we are trying to change already has a value - just return to not overwrite it
+    if (valuesArr[i]) {
+      return;
+    }
+
+    // If the game is over, return so that the squares can't be clicked anymore
+    if (winner) {
+      return;
+    }
+
+    const valuesArrClone = valuesArr.slice(); // Same as [...valuesArr] to make the state immutable
+    valuesArrClone[i] = nextPlayer; // nextPlayer is the next move so we just update the index with that
+    setValuesArr(valuesArrClone); // To update the valuesArr
+    setNextPlayer(nextPlayer === "O" ? "X" : "O"); // To swap the next player to X or back to O
+  }
+
+  function checkWinner() {
```

Comment:
`checkWinner` gets called every `rerender` which is okay in this case. But if we have a bigger project with different components and maybe a 100x bigger board. Then this will introduce a performance issue because it will compute every time.

A better way to do this is to just call `checkWinner` whenever `updateValuesArr` is called because that's the time where a winner might come up.

Also, I saw that you returned `winner` here. Once you follow the flow above, you can just pass `winner` as a prop to `Board` then display it as is.